src/cloud_radar/cf/unit/_template.py

```diff
@@ -250,9 +280,74 @@ def resolve_values(
                 )
                 for item in data
             ]
+        elif isinstance(data, str):
+            return self.resolve_dynamic_references(data)
         else:
             return data
 
+    def resolve_dynamic_references(self, data: str) -> str:
+        """
+        Replaces any dynamic references in the provided data with values from
+        our configuration.
+
+        Args:
+            data (str): the value to replace dynamic references within
+
+        Raises:
+            ValueError: If a dynamic reference has been used and no dynamic
+            references have been configured.
+            KeyError: If a dynamic reference name has been used and is not
+            found in the configuration
+        """
+
+        if "{{resolve:" in data:
```

Comment:
Changing this line to `if self.dynamic_references and "{{resolve:" in data:` would make this not a breaking change, so if `dynamic_references` was completely empty none of the resolution logic would be performed

Reply:
Breaking change seems like the right call.